components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/DAOUtil.java

```diff
@@ -105,5 +107,68 @@ public static String getConvertedAutoGeneratedColumnName(String dbProductName, S
     public static void clearDataSource() {
         dataSource = null;
     }
+
+    /**
+     * Utility method to close the connection streams.
+     *
+     * @param preparedStatement PreparedStatement
+     * @param connection        Connection
+     * @param resultSet         ResultSet
+     */
+    public static void closeAllConnections(PreparedStatement preparedStatement, Connection connection,
+            ResultSet resultSet) {
+        closeConnection(connection);
+        closeResultSet(resultSet);
+        closeStatement(preparedStatement);
+    }
+
+    /**
+     * Close Connection.
+     *
+     * @param dbConnection Connection
+     */
+    private static void closeConnection(Connection dbConnection) {
+        if (dbConnection != null) {
+            try {
+                dbConnection.close();
+            } catch (SQLException e) {
+                log.warn("Database error. Could not close database connection. Continuing with " + "others. - " + e
+                        .getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Close ResultSet.
+     *
+     * @param resultSet ResultSet
+     */
+    private static void closeResultSet(ResultSet resultSet) {
+        if (resultSet != null) {
+            try {
+                resultSet.close();
+            } catch (SQLException e) {
+                log.warn("Database error. Could not close ResultSet  - " + e.getMessage(), e);
+            }
+        }
+
+    }
+
+    /**
+     * Close PreparedStatement.
+     *
+     * @param preparedStatement PreparedStatement
+     */
+    public static void closeStatement(PreparedStatement preparedStatement) {
```

Comment:
Use try catch with resources pattern to avoid the need of explicitly closing resources [1].

[1]. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html